dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentTypeLoader.java

```diff
@@ -0,0 +1,303 @@
+                    .append("ShortFormat =\"[ ")
+                    .append(field.variable())
+                    .append(" ]\")");
+                sb.append("#set( $")
+                    .append(field.variable())
+                    .append("LongFormat =\"[ ")
+                    .append(field.variable())
+                    .append(" ]\")");
+                sb.append("#set( $")
+                    .append(field.variable())
+                    .append("DBFormat =\"[ ")
+                    .append(field.variable())
+                    .append(" ]\")");
+            }
+        }
+
+        // sets the categories as a list on velocity
+        sb.append("#set( $ContentletCategories =[] )");
+        sb.append("#set( $ContentletCategoryNames =[] )");
+
+
+        // This is code is repeated becuase the bug GETTYS-268, the content variables were been
+        // overwritten
+        // by the parse inside the some of the content fields
+        // To edit the look, see WEB-INF/velocity/static/preview/content_controls.vtl
+
+        sb.append("#set( $EDIT_CONTENT_PERMISSION =$EDIT_CONTENT_PERMISSION")
+            .append(fakeIdentifier)
+            .append(" )");
+        sb.append("#set( $CONTENT_INODE ='")
+            .append(fakeInode)
+            .append("' )");
+        sb.append("#set( $IDENTIFIER_INODE ='")
+            .append(fakeIdentifier)
+            .append("' )");
+
+        sb.append("#set( $ContentInode ='")
+            .append(fakeInode)
+            .append("' )");
+        sb.append("#set( $ContentIdentifier ='")
+            .append(fakeIdentifier)
+            .append("' )");
+        sb.append("#set( $ContentletTitle =\"")
+            .append(UtilMethods.espaceForVelocity(fakeTitle))
+            .append("\" )");
+        sb.append("#set( $isWidget = false)");
+
+        return writeOutVelocity(filePath, sb.toString());
+    }
+
+    public void removeContentTypeFile(ContentType contentType) {
+
+    }
+
+    @Override
+    public InputStream writeObject(String id1, String id2, PageMode mode, String language, String filePath)
+            throws DotDataException, DotSecurityException {
+
+        ContentTypeAPI contentTypeAPI = APILocator.getContentTypeAPI(sysUser());
+
+
+        // Search for the given ContentType inode
+        ContentType foundContentType = contentTypeAPI.find(id1);
+
+
+        return buildVelocity(foundContentType,  filePath, mode);
+
+    }
+
+
+
+    @Override
+    public void invalidate(Object obj, PageMode mode) {
+        ContentType contentType = (ContentType)obj;
+        String folderPath =  mode.name() + File.separator;
+        String filePath = folderPath + contentType.inode() + "." + VelocityType.CONTENT_TYPE.fileExtension;
+
+        String velocityRootPath = VelocityUtil.getVelocityRootPath();
+        String absolutPath = velocityRootPath + File.separator + filePath;
+        java.io.File f = new java.io.File(absolutPath);
+        f.delete();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use "java.nio.Files#delete" here for better messages on error conditions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Do something with the "boolean" value returned by "delete". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS899)